Uyuni proxy #10177 ke (#45)
uyuni-project/uyuni-docs: modules/installation/pages/uyuni-proxy-registration.adoc

channels by clicking the ``include recommended`` icon:
+
* SLE-Module-Basesystem15-SP1-Pool
* SLE-Module-Basesystem15-SP1-Updates
* SLE-Module-Server-Applications15-SP1-Pool
* SLE-Module-Server-Applications15-SP1-Updates
* SLE-Module-SUSE-Manager-Proxy-4.0-Pool
* SLE-Module-SUSE-Manager-Proxy-4.0-Updates
+
The [systemitem]``SLE-Product-SUSE-Manager-Proxy-4.0-Updates`` channel is mandatory.
+

// .Base and Child Proxy Channel
// image::proxy-child.png[]
////

. OPTIONAL: Modify a bootstrap script for the proxy.
If you want to run the proxy on a traditional client (system type ``Management``) uncheck [guimenu]``Bootstrap using Salt``.
Using Salt is the default.
// What's up with:
// Enable Remote Configuration
// Enable Remote Commands
For more information about bootstrap scripts, see
xref:client-configuration:registration-bootstrap.adoc[].
+

// .Modifying Bootstrap Script
// image::proxy-bootstrap.png[]

. Create the SUSE Manager Tools Repository for bootstrapping, see xref:client-configuration:creating-a-tools-repository.adoc[Create Tools Repository].
. Bootstrap the client using the script.
For more information, see xref:client-configuration:registration-bootstrap.adoc[].

> . For Salt clients, accept the key on the menu:Salt[Keys] page by checking the appropriate checkbox.

Author: ```suggestion
. For Salt clients, navigate to menu:Salt[Keys] and accept the key.
```

[[proxy-register]]
= Proxy Registration

Proxy systems are registered as Salt clients using a bootstrap script.

> This procedure describes software channel setup and registering the installed Proxy with an activation key as a {productname} client.

Reviewer: ```suggestion
This procedure describes software channel setup and registering the installed proxy with an activation key as a {productname} client.
```

[IMPORTANT]

> .Downloading Channels

Reviewer: ```suggestion
```

====
Before you can select the correct child channels while creating the activation key, ensure you have created an {productname} Proxy 4 channel with all needed repositories properly synchronized.
====

[[proxy-register-procedure]]
.Procedure: Registering the Proxy

. Create an activation key based on your {productname} Proxy 4 channel.
For more information about activation keys, see xref:client-configuration:clients-and-activation-keys.adoc[Creating Activation Key].
+

// .Proxy Activation Key
// image::proxy-key.png[]
+

> ////

Reviewer 2: Shouldn't we remove this is (and other comments) if this doc is only for Uyuni?

Author: in general, yes.  But here we must check, which channel arrangement is actually needed (or the best).

. From the [guimenu]``Child Channels`` listing select the recommended
channels by clicking the ``include recommended`` icon:
+
* SLE-Module-Basesystem15-SP1-Pool
* SLE-Module-Basesystem15-SP1-Updates
* SLE-Module-Server-Applications15-SP1-Pool
* SLE-Module-Server-Applications15-SP1-Updates
* SLE-Module-SUSE-Manager-Proxy-4.0-Pool
* SLE-Module-SUSE-Manager-Proxy-4.0-Updates
+
The [systemitem]``SLE-Product-SUSE-Manager-Proxy-4.0-Updates`` channel is mandatory.
+

// .Base and Child Proxy Channel
// image::proxy-child.png[]
////

> . Modify a bootstrap script for the proxy if needed.

Reviewer: ```suggestion
. OPTIONAL: Modify a bootstrap script for the proxy.
```

If you want to run the proxy on a traditional client (system type ``Management``) uncheck [guimenu]``Bootstrap using Salt``.
Using Salt is the default.
// What's up with:
// Enable Remote Configuration
// Enable Remote Commands
For more information about bootstrap scripts, see
xref:client-configuration:registration-bootstrap.adoc[].
+

// .Modifying Bootstrap Script
// image::proxy-bootstrap.png[]

. Create the SUSE Manager Tools Repository for bootstrapping, see xref:client-configuration:creating-a-tools-repository.adoc[Create Tools Repository].

> . Bootstrap the client with the bootstrap script.

Reviewer: ```suggestion
. Bootstrap the client using the script.
```

For more information, see xref:client-configuration:registration-bootstrap.adoc[].

> . For Salt clients, accept the key on the menu:Salt[Keys] page by checking the appropriate checkbox.

Reviewer: Word order.

Author: done

> When accepted, it will appear in the menu:Systems[Overview].

Reviewer: ```suggestion
When the key is accepted, it will show in menu:Systems[Overview].
```

. Navigate to menu:System Details[Software > Software Channels], and check that your proxy channel is selected.
// [systemitem]``SLE-PRODUCT-Pool`` must be the base channel and the others are child channels.
+

// .Proxy Channels
// image::sles15-proxy-channels.png[]

> For setting up the registered {productname} Proxy, see xref:uyuni-proxy-setup.adoc[].

Reviewer: ```suggestion
For more on setting up a registered {productname} Proxy, see xref:uyuni-proxy-setup.adoc[].
```

[[proxy-register]]
= Proxy Registration

Proxy systems are registered as Salt clients using a bootstrap script.

This procedure describes software channel setup and registering the installed proxy with an activation key as a {productname} client.

[IMPORTANT]
====
Before you can select the correct child channels while creating the activation key, ensure you have properly synchronized the openSUSE Leap channel with all the needed child channels and the {productname} Proxy channel.
====

> // ==>>>> uyuni-proxy-stable-leap-151 channel

Reviewer 2: Why is this comment here?

[[proxy-register-procedure]]
.Procedure: Registering the Proxy

. First create openSUSE Leap and the {productname} Proxy channels with the [command]``spacewalk-common-channels`` command.
[command]``spacewalk-common-channels`` is part of the [package]``spacewalk-utils`` package:
+
----
spacewalk-common-channels \
opensuse_leap15_1 \
opensuse_leap15_1-non-oss \
opensuse_leap15_1-non-oss-updates \
opensuse_leap15_1-updates \
opensuse_leap15_1-uyuni-client \
uyuni-proxy-stable-leap-151
----
+
For more information, see xref:client-configuration:clients-opensuse.adoc[].

. Create the openSUSE Leap 15 Uyuni Client Tools Repository for bootstrapping, see xref:client-configuration:creating-a-tools-repository.adoc[].
. Create an activation key with openSUSE Leap as a base channel and the other channels as child channels.
For more information about activation keys, see xref:client-configuration:clients-and-activation-keys.adoc[].
+

> // // .Proxy Activation Key

Reviewer 2: Shouldn't we remove this, as it doesn't apply for Uyuni?

// // image::proxy-key.png[]
// +
// ////
// . From the [guimenu]``Child Channels`` listing select the recommended
// channels by clicking the ``include recommended`` icon:
// +
// * SLE-Module-Basesystem15-SP1-Pool
// * SLE-Module-Basesystem15-SP1-Updates
// * SLE-Module-Server-Applications15-SP1-Pool
// * SLE-Module-Server-Applications15-SP1-Updates
// * SLE-Module-SUSE-Manager-Proxy-4.0-Pool
// * SLE-Module-SUSE-Manager-Proxy-4.0-Updates
// +
// The [systemitem]``SLE-Product-SUSE-Manager-Proxy-4.0-Updates`` channel is mandatory.
// +
// .Base and Child Proxy Channel
// image::proxy-child.png[]

. Modify a bootstrap script for the proxy.
* Add the GPG key to the [systemitem]``ORG_GPG_KEY=`` parameter.
For more information, see xref:client-configuration:clients-opensuse.adoc[].
* OPTIONAL: If you want to run the proxy on a traditional client (system type ``Management``) uncheck [guimenu]``Bootstrap using Salt``.
Using Salt is the default.
+

> // FIXME: I do not know whether changing to traditional is actually possible

Reviewer 2: FMPOV on Uyuni we should support only proxies based on salt clients. @paususe @mcalmer?

If so, this could be removed.

+
// What's up with:
// Enable Remote Configuration
// Enable Remote Commands
+
For more information about bootstrap scripts, see
xref:client-configuration:registration-bootstrap.adoc[].
+

> // .Modifying Bootstrap Script

Reviewer 2: Why do we need this commented?

// image::proxy-bootstrap.png[]

. Bootstrap the client using the script.
For more information, see xref:client-configuration:registration-bootstrap.adoc[].
. For Salt clients, navigate to menu:Salt[Keys] and accept the key.
When the key is accepted, it will show in menu:Systems[Overview].
. Navigate to menu:System Details[Software > Software Channels], and check that your proxy channel is selected.
+

> // .Proxy Channels

Reviewer 2: Why do we need this commented?
